yarn-project/prover-client/src/proving_broker/caching_broker_facade.ts

```diff
@@ -147,6 +147,8 @@ export class CachingBrokerFacade implements ServerCircuitProver {
       }
     } finally {
       signal?.removeEventListener('abort', abortFn);
+      // we've saved the result in our cache. We can tell the broker to clear its state
+      await this.broker.cleanUpProvingJobState(id);
```

Comment:
If I read this correctly. Does this mean the orchestrator effectively tells the broker when to clear state. This feels like it's quite coupled. Would a better approach be for the broker to simply e.g. delete all state for epochs < N - 1 when it is asked to prove something for epoch N.

Reply:
Yes, you that's right. I think your suggestion makes a lot of sense!